backend/tests/integration/controllers/posts-controller.test.js

```diff
@@ -0,0 +1,155 @@
+  });
+
+  it('Post creation: Failure - Invalid image URL format', async () => {
+    const postObject = createPostObject({
+      imageLink: 'https://www.invalid-image.gif',
+    });
+    const response = await request(server).post('/api/posts').send(postObject);
+
+    expect(JSON.parse(response.text)).toEqual({
+      message: 'Image URL must end with .jpg, .jpeg, or .png',
+    });
+    expect(response.status).toBe(400);
+  });
+
+  it('Post creation: Failure - Internal server error', async () => {
+    // Mocking a scenario where the server encounters an internal error during post creation
+    jest.spyOn(Post.prototype, 'save').mockRejectedValueOnce(new Error('Internal Server Error'));
+
+    const postObject = createPostObject();
+    const response = await request(server).post('/api/posts').send(postObject);
+
+    expect(response.status).toBe(500);
+    expect(JSON.parse(response.text)).toEqual({ message: 'Internal Server Error' });
+  });
+});
+describe('Integration Tests: Get all posts', () => {
+  it('Get all posts: Success', async () => {
+    const response = await request(server).get('/api/posts');
+
+    expect(response.status).toBe(200);
+    expect(response.body).toBeInstanceOf(Array);
+  });
+});
+describe('Integration Tests: Get all posts by category', () => {
+  it('Get all posts by category: Success', async () => {
+    const category = validCategories[0];
+    const response = await request(server).get(`/api/posts/categories/${category}`);
+
+    expect(response.status).toBe(200);
+    expect(response.body).toBeInstanceOf(Array);
+  });
+
+  it('Get all posts by category: Failure - Invalid category', async () => {
+    const category = 'invalid-category';
+    const response = await request(server).get(`/api/posts/categories/${category}`);
+
+    expect(response.status).toBe(400);
+    expect(JSON.parse(response.text)).toEqual({
+      message: 'Invalid category',
+    });
+  });
+});
+describe('Integration Tests: Get all featured posts', () => {
+  it('Get all featured posts: Success', async () => {
+    const responseFeatured = await request(server).get('/api/posts/featured');
+
+    expect(responseFeatured.status).toBe(200);
+    expect(responseFeatured.body).toBeInstanceOf(Array);
+  });
+});
+describe('Integration Tests: Get all latest posts', () => {
+  it('Get all latest posts: Success', async () => {
+    const responseLatest = await request(server).get('/api/posts/latest');
+
+    expect(responseLatest.status).toBe(200);
+    expect(responseLatest.body).toBeInstanceOf(Array);
+  });
+});
+describe('Integration Tests: Update Post', () => {
+  it('Update Post: Success - Update Post of existing ID', async () => {
+    const response = await request(server)
+      .patch(`/api/posts/${postId}`)
+      .send(createPostObject({ title: 'Updated Post' }));
+
+    expect(response.status).toBe(200);
+    expect(response.body).toHaveProperty('_id', postId);
+    const updatedPost = await Post.findById(postId);
```

Comment:
LOC 121-123 are unncessary.
Better compare before and previous expected records 
for clarity

Reply:
> LOC 121-123 are unncessary.

Unsure about that, as I am testing if the title is updated with my changed proposed title.
p.s: I'll apply AAA for this.